src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1270,19 +1270,30 @@ private string[] SplitInternal(ReadOnlySpan<char> separators, int count, StringS
                 throw new ArgumentOutOfRangeException(nameof(count),
                     SR.ArgumentOutOfRange_NegativeCount);
 
-            if (options < StringSplitOptions.None || options > StringSplitOptions.RemoveEmptyEntries)
-                throw new ArgumentException(SR.Format(SR.Arg_EnumIllegalVal, options));
+            CheckStringSplitOptions(options);
 
-            bool omitEmptyEntries = (options == StringSplitOptions.RemoveEmptyEntries);
-
-            if ((count == 0) || (omitEmptyEntries && Length == 0))
+        ShortCircuit:
+            if (count <= 1 || Length == 0)
             {
-                return Array.Empty<string>();
+                // Per the method's documentation, we'll short-circuit the search for separators.
+                // But we still need to post-process the results based on the caller-provided flags.
+
+                string candidate = this;
```

Comment:
Is it worth checking for an empty input string now, in case both TrimEntries and RemoveEmptyEntries is set?

Reply:
Yup! That's actually already done on line 1276.

```cs
        ShortCircuit:
            if (count <= 1 || Length == 0)  // <-- empty input string causes this to be true
            {
                // Per the method's documentation, we'll short-circuit the search for separators.
                // But we still need to post-process the results based on the caller-provided flags.
```